### ISSUE TITLE
Move nighthawk onto architecture that supports envoy proto extension lookup by type, not name

### ISSUE THREAD

**Issue by dubious90** [envoyproxy/nighthawk#829]
*Description*:
We have added in "envoy.reloadable_features.no_extension_lookup_by_name=false" in many places.  However, this is a short term solution, and need to adjust our architecture to support lookup by type instead of name, particularly in the nighthawk test server.

See Envoy PR https://github.com/envoyproxy/envoy/pull/20397 for more information.

**Comment by dubious90** [comment]
@mum4k Do you have any details here on what would need to happen here?

**Comment by mum4k** [comment]
Thank you for filing this @dubious90.

TLDR; we have three test server extensions and they all share the same configuration type [ResponseOptions](https://github.com/envoyproxy/nighthawk/blob/5f72e3d9315c0fb152f65c1c154ee4beb96ceb52/api/server/response_options.proto#L22). This used to work well when Envoy was looking up extensions by name, but the recent change switched that to the message type which now has to be unique.

To be said, we also accept the same type `ResponseOptions` when encoded in HTTP request headers, the test server receives runtime configuration this way.

One option would be to define three new messages and each of these messages will wrap the current `ResponseOptions`, i.e. we move it a level down. Registration and lookup of extensions will work based on these new unique types, while they will continue getting configuration from the same (now inner) message. We can continue sending encoded `ResponseOptions` in headers without change. Open to other ideas.

/cc @oschaaf in case Otto has better ideas.

**Comment by mum4k** [comment]
Fyi the workaround was implemented by setting `envoy.reloadable_features.no_extension_lookup_by_name` = `False` in #828.
